### PR TITLE
Packaging for release v1.4.0

### PR DESCRIPTION
**Pull request by paulomarg**
Release v1.4.0 of the CLI, with improved debugging for errors in Partners API requests.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+Version 1.4.0
+------
+* Updates to test, dependencies and internal tooling
```
Comment (andyw8) [inline CHANGELOG.md:3]: Should we start a habit of linking to the PRs?

Comment (andyw8) [reply CHANGELOG.md:3]: (also s/test/tests ?)

**Review by andyw8 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (mkevinosullivan) [reply CHANGELOG.md]: It probably would be useful; we just need to be cognizant that some of the stuff contained within App CLI is not quite fully public yet (e.g., Argo-related).

Comment (paulomarg) [reply CHANGELOG.md]: I'm tweaking the PR template to add changelog updates as a checkbox so we keep track of the relevant information as it comes in, so we can remind contributors to consider whether it's a change worth mentioning.

**Pull request merged**